e2e/tests/notebook/reader-copy-to-workspace.spec.ts

```diff
@@ -0,0 +1,143 @@
+import Navigation, {NavLink} from 'app/component/navigation';
+import Link from 'app/element/link';
+import DataPage, {TabLabelAlias} from 'app/page/data-page';
+import WorkspaceAboutPage from 'app/page/workspace-about-page';
+import WorkspaceAnalysisPage from 'app/page/workspace-analysis-page';
+import {EllipsisMenuAction, Language, LinkText, WorkspaceAccessLevel} from 'app/text-labels';
+import {config} from 'resources/workbench-config';
+import {makeRandomName} from 'utils/str-utils';
+import {findWorkspace, signIn, signInAs, waitWhileLoading} from 'utils/test-utils';
+import WorkspaceCard from 'app/component/workspace-card';
+import DataResourceCard, {CardType} from 'app/component/data-resource-card';
+import Modal from 'app/component/modal';
+import NotebookPreviewPage from 'app/page/notebook-preview-page';
+
+jest.setTimeout(20 * 60 * 1000);
+
+describe('Workspace reader Jupyter notebook action tests', () => {
+
+  beforeEach(async () => {
+    await signIn(page);
+  });
+
+  /**
+   * Test: Workspace reader can copy notebook to another Workspace then gain Edit right to the clone notebook.
+   * - Create a Workspace and a new notebook. Run simple code in notebook and save notebook.
+   * - Share Workspace with collaborator as READER.
+   * - Sign in as collaborator.
+   * - Create Workspace where collaborator is the OWNER.
+   * - Verify Workspace and notebook permissions.
+   * - Copy notebook to collaborator Workspace.
+   * - Full Edit rights to clone notebook.
+   * - Verify original notebook contents are unchaged in clone notebook.
+   * - Delete clone notebook.
+   */
+  test('Workspace reader copy notebook to another workspace', async () => {
+    const workspaceName = await findWorkspace(page, {create: true}).then(card => card.clickWorkspaceName());
+
+    let dataPage = new DataPage(page);
```

Comment:
Same basic premise as above - could the `let` variables be factored out?

Reply:
got rid of all `let`.